android/src/main/java/com/tailscale/ipn/App.java

```diff
@@ -372,4 +379,267 @@ String getInterfacesAsString() {
+		if (networks == null) {
+			// Android 6 and before often returns an empty list, but we
+			// can try again with just the active network.
+			//
+			// Once Tailscale is connected, the active network will be Tailscale
+			// which will have 100.100.100.100 for its DNS server. We reject
+			// TYPE_VPN in getPreferabilityForNetwork, so it won't be returned.
+			Network active = cMgr.getActiveNetwork();
+			if (active == null) {
+				return "";
+			}
+			networks = new Network[]{active};
+		}
+
+		String[] dnsConfigs = new String[]{"", "", "", ""};
+		for (Network network : networks) {
+			int idx = getPreferabilityForNetwork(cMgr, network);
+			if (idx < 0) {
+				continue;
+			}
+
+			LinkProperties linkProp = cMgr.getLinkProperties(network);
+			NetworkCapabilities nc = cMgr.getNetworkCapabilities(network);
+			List<InetAddress> dnsList = linkProp.getDnsServers();
+			StringBuilder sb = new StringBuilder("");
+			for (InetAddress ip : dnsList) {
+				sb.append(ip.getHostAddress() + " ");
+			}
+
+			String d = linkProp.getDomains();
+			if (d != null) {
+				sb.append("\n");
+				sb.append(d);
+			}
+
+			dnsConfigs[idx] = sb.toString();
+		}
+
+		// return the lowest index DNS config which exists. If an Ethernet config
+		// was found, return it. Otherwise if Wi-fi was found, return it. Etc.
+		for (String s : dnsConfigs) {
+			if (!s.trim().isEmpty()) {
+				return s;
+			}
+		}
+
+		return "";
+	}
+
+	// getDnsServersFromSystemProperties returns DNS servers found in system properties.
+	// On Android versions prior to Android 8, we can directly query the DNS
+	// servers the system is using. More recent Android releases return empty strings.
+	//
+	// Once Tailscale is connected these properties will return 100.100.100.100, which we
+	// suppress.
+	//
+	// on a Nexus 4  with Android  5.1 on wifi: 2602:248:7b4a:ff60::1 10.1.10.1
+	// on a Nexus 7  with Android  6.0 on wifi: 2602:248:7b4a:ff60::1 10.1.10.1
+	// on a Pixel 3a with Android 12.0 on wifi:
+	// on a Pixel 3a with Android 12.0 on  LTE:
+	//
+	// The list of DNS search domains does not appear to be available in system properties.
+	String getDnsServersFromSystemProperties() {
+		try {
+			Class SystemProperties = Class.forName("android.os.SystemProperties");
+			Method method = SystemProperties.getMethod("get", String.class);
+			List<String> servers = new ArrayList<String>();
+			for (String name : new String[]{"net.dns1", "net.dns2", "net.dns3", "net.dns4"}) {
+				String value = (String) method.invoke(null, name);
+				if (value != null && !value.isEmpty() &&
+						!value.equals("100.100.100.100") &&
+						!servers.contains(value)) {
+					servers.add(value);
+				}
+			}
+			return String.join(" ", servers);
+		} catch (Exception e) {
+			return "";
+		}
+	}
+
+
+	String intToInetString(int hostAddress) {
+		return String.format("%d.%d.%d.%d", (byte)(0xff & hostAddress),
+			(byte)(0xff & (hostAddress >> 8)), (byte)(0xff & (hostAddress >> 16)),
```

Comment:
put the >> 16 expression on its own line (and the first one too without the shift) so they all line up

Reply:
Done.